src/stores/messages/MessagesDataProviderStore.ts

```diff
@@ -197,8 +198,15 @@ export default class MessagesDataProviderStore implements MessagesDataStore {
 
 		runInAction(() => {
 			const messages = [
-				...nextMessages.filter(val => val.messageId !== message?.messageId),
+				...nextMessages.filter(
```

Comment:
so i guess the problem is the middle message that we put in center manually.
the way you can solve it here is just by getting unique messages, and than sorting them, i think that would work.

but you also have to solve the same problem in onNextChannelResponse
and i'm not sure how to do that